### ISSUE TITLE
Title / Share function

### ISSUE THREAD

**Issue by Hirbod** [sarriaroman/photoviewer#3]
Finally someone managed to develop a working solution for both iOS and Android with Remote Image support. And no picture intent on android. :+1:  :+1:  :100:  You can't imagine how thankful I am!

Just two small things:

Title for image and Gallery modus (array of images)
UIDocumentController support title, just a small adjustment, but I don't know if Andorid support's title

**Comment by sarriaroman** [comment]
About the two things:
- Title: yes, it's absolutely possible.
- Gallery: I didn't thought the plugin to work in that way and the change it's a bit harder.

**Comment by Hirbod** [comment]
If you find some time.. I would post a bounty on it, but maybe we would find some more people to make a donation for that feature.

By the way, Title should be optional.

**Comment by Hirbod** [comment]
Sorry, for the spam: UIDocumentController gives you the ability to save, share etc.
Is it possible to add a (optional) save button on android? Maybe with a long-press-gesture?

**Comment by sarriaroman** [comment]
The feature to handle a set of images can be planned. Maybe moving it to a different issues.

The title will be optional. I agree with that.

**Comment by sarriaroman** [comment]
Yes, It's possible with a long press gesture. I can work on both, the share and title ;)

**Comment by Hirbod** [comment]
:+1: Thank you.

**Comment by sarriaroman** [comment]
If you are working over a project you can try pulling from master for both features. I made a push.

**Comment by Hirbod** [comment]
Wow, you're fast!!!

**Comment by Hirbod** [comment]
iOS work fine, Android does not work:

```bash
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:20: error: cannot find symbol
	final GestureDetector gestureDetector = new GestureDetector(new GestureDetector.SimpleOnGestureListener() {
	      ^
  symbol:   class GestureDetector
  location: class PhotoActivity
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:61: error: cannot find symbol
	public boolean onTouchEvent(MotionEvent event) {
	                            ^
  symbol:   class MotionEvent
  location: class PhotoActivity
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:20: error: cannot find symbol
	final GestureDetector gestureDetector = new GestureDetector(new GestureDetector.SimpleOnGestureListener() {
	                                            ^
  symbol:   class GestureDetector
  location: class PhotoActivity
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:20: error: package GestureDetector does not exist
	final GestureDetector gestureDetector = new GestureDetector(new GestureDetector.SimpleOnGestureListener() {
	                                                                               ^
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:37: error: cannot find symbol
		actTitle = this.getIntent().getStringExtra("title");
		^
  symbol:   variable actTitle
  location: class PhotoActivity
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:38: error: cannot find symbol
		if( !actTitle.equals("") ) {
		     ^
  symbol:   variable actTitle
  location: class PhotoActivity
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:39: error: cannot find symbol
			this.setTitle(actTitle);
			              ^
  symbol:   variable actTitle
  location: class PhotoActivity
```

**Comment by sarriaroman** [comment]
Ups... I forgot the imports. Sorry! Added now!

Thanks for you testing :)

**Comment by Hirbod** [comment]
Nope.. still

```bash
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:66: error: cannot find symbol
	public boolean onTouchEvent(MotionEvent event) {
	                            ^
  symbol:   class MotionEvent
  location: class PhotoActivity
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:25: error: cannot find symbol
	    public void onLongPress(MotionEvent e) {
	                            ^
  symbol: class MotionEvent
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:29: error: cannot find symbol
			sharingIntent.putExtra(Intent.EXTRA_STREAM, Uri.parse(imageUrl));
			                                            ^
  symbol: variable Uri
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:31: error: cannot find symbol
			startActivity(Intent.createChooser(shareIntent, "Share"));
			                                   ^
  symbol: variable shareIntent
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:41: error: cannot find symbol
		actTitle = this.getIntent().getStringExtra("title");
		^
  symbol:   variable actTitle
  location: class PhotoActivity
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:42: error: cannot find symbol
		if( !actTitle.equals("") ) {
		     ^
  symbol:   variable actTitle
  location: class PhotoActivity
/Users/Hirbod/Desktop/appdev/Nachbarschaft/platforms/android/src/com/speryans/PhotoViewer/PhotoActivity/PhotoActivity.java:43: error: cannot find symbol
			this.setTitle(actTitle);
			              ^
  symbol:   variable actTitle
  location: class PhotoActivity
```

**Comment by sarriaroman** [comment]
Note that I'm editing in a text editor hehe.

**Comment by Hirbod** [comment]
:))

**Comment by sarriaroman** [comment]
Done!

**Comment by Hirbod** [comment]
It works on iOS, and build does not fail on Android anymore, but now I get:

Uncaught ReferenceError: PhotoViewer is not defined

**Comment by sarriaroman** [comment]
I will check in a project because for some reason is not getting the plugin on Android. Did you try to remove the plugin and add it again?

**Comment by Hirbod** [comment]
yes, I also removed my android platform.

**Comment by Hirbod** [comment]
also tried to download from npm and directly from git

**Comment by Hirbod** [comment]
Ok, Problem caused trough

cordova-hot-code-push-plugin

I removed the plugin. Now it works.

**Comment by sarriaroman** [comment]
That's awesome!

**Comment by Hirbod** [comment]
Weird.. as soon as I install cordova-hot-code-push-plugin, your plugin fail to work. Before I updated everything worked fine.. what happened?

**Comment by sarriaroman** [comment]
I'm reading about the plugin, but the main change that can break it is the domain of the plugin (Plugin ID) updated for NPM.

**Comment by Hirbod** [comment]
iOS works fine.. I guess there is some cache I need to clear somehow..

**Comment by sarriaroman** [comment]
Please remove the plugin and install the new version. I think that is the plugin id

**Comment by Hirbod** [comment]
ok

**Comment by Hirbod** [comment]
Yes, it works, even with cordova-hot-code-push-plugin :+1: 
Thanks. But why does the plugin id make problems?

**Comment by Hirbod** [comment]
I dont see any title on android, long-press-gesture also does not work

**Comment by sarriaroman** [comment]
Maybe the cordova-hot-code-push-plugin can't accept plugins with dashes in the id ;).

**Comment by Hirbod** [comment]
No, I dont think so, I have all the cordova-plugin-file etc installed.
Nevermind, it works now, but as said, title and gesture does not work

**Comment by sarriaroman** [comment]
The long press seems to be related with the PhotoView library that handle the zoom feature. I will go deep on this problem. I'm checking in a project first to update it again.

**Comment by sarriaroman** [comment]
Implemented on version 1.1.1 with a new Title/Share bar.

**Comment by Hirbod** [comment]
Confirmed! Looks great and works perfectly. Sadly there is no "save to photos" option, but I think that is ok.
